src/diffpy/labpdfproc/functions.py

```diff
@@ -208,7 +205,40 @@ def compute_cve(diffraction_data, mud, wavelength):
     muls = np.array(muls) / abs_correction.total_points_in_grid
     cve = 1 / muls
 
+    abdo_resampled = Diffraction_object(wavelength=wavelength)
+    abdo_resampled.insert_scattering_quantity(
+        TTH_GRID,
+        cve,
+        "tth",
+        name=f"absorption correction, cve, for mud {mud}",
+        wavelength=wavelength,
+        scat_quantity="cve",
+    )
+
+    return abdo_resampled
+
+
+def interpolate_cve(diffraction_data, abdo_resampled, wavelength):
+    """
+    interpolate cve to the original grid in diffraction_data
+
+    Parameters
+    ----------
+    diffraction_data Diffraction_object
+      the diffraction pattern
+    abdo_resampled Diffraction_object
+      the diffraction object with cve curves on the resampled grid
+    wavelength float
+      the wavelength of the diffraction object
+
+    Returns
+    -------
+    the diffraction object with cve curves
+
+    """
+
     orig_grid = diffraction_data.on_tth[0]
+    cve = abdo_resampled.on_tth[1]
     newcve = np.interp(orig_grid, TTH_GRID, cve)
```

Comment:
this would be easier to read if, instead of `TTH_GRID` you used `abdo_resampled.on_tth[0]`.  Actually, all in all, I think it would be easier to read as:
```
cve_on_data_grid = np.interp(diffraction_data.on_tth[0], cve.on_tth[0], cve.on_tth[1])
```